src/sentry/api/endpoints/organization_events_stats.py

```diff
@@ -453,20 +454,30 @@ def fn(
         get_event_stats = get_event_stats_factory(dataset)
 
         try:
-            return Response(
-                self.get_event_stats_data(
-                    request,
-                    organization,
-                    get_event_stats,
-                    top_events,
-                    allow_partial_buckets=allow_partial_buckets,
-                    zerofill_results=not (
-                        request.GET.get("withoutZerofill") == "1" and has_chart_interpolation
-                    ),
-                    comparison_delta=comparison_delta,
-                    dataset=dataset,
+            response_data = self.get_event_stats_data(
+                request,
+                organization,
+                get_event_stats,
+                top_events,
+                allow_partial_buckets=allow_partial_buckets,
+                zerofill_results=not (
+                    request.GET.get("withoutZerofill") == "1" and has_chart_interpolation
                 ),
+                comparison_delta=comparison_delta,
+                dataset=dataset,
+            )
+            resp = Response(
+                response_data,
                 status=200,
             )
+            if (
+                response_data.get("end")
+                and request.GET.get("end")
+                and request.GET.get("project")  # only works if projects are explicitly specified
+                and time.time() - response_data.get("end") > 7200
```

Comment:
Could we explicitly check this isn't a statsPeriod query for a small period?

Reply:
oh nvm i misread this, its checking that the end of the query is >2hr ago right?